### ISSUE TITLE
Cannot create mongo repositories for @Value.Include

### ISSUE THREAD

**Issue by scottbessler** [immutables/immutables#203]
It appears that `@Mongo.Repository` only works for `@Value.Immutable`, so if I want my interfaces in one library and my Immutable classes and their repositories in another I appear to be stuck.

Am I missing something?

**Comment by elucash** [comment]
We don't currently fully support such use cases. However, here are some things that might work

1) Have interfaces in one package/library and extend/implement it in another package using immutable implementation using intermediate abstract value type. Here's an example

```java
package interfaces;
// fully abstract definition
public interface ValueObject {
   String name();
   int count();
}

package implementations;
...
@Value.Immutable
@Mongo.Repository
public interface ValueObject extends interfaces.ValueObject {}

...

// use generated
implementations.ImmutableValueObject
implementations.ValueObjectRepository 
```

Then use styles to suite naming, visibility levels and other conventions to suit your codebase: http://immutables.github.io/style.html

2) Currently we have some support of abstract value types defined in external package by using  `Value.Include` annotation: http://immutables.github.io/immutable.html#value . Unfortunately this will not work for repositories, but I might take a look if we could extend it somehow to cover additional cases for `Repository` etc.

Let us know which option might work for you. Thank you for the issue report!

**Comment by scottbessler** [comment]
Yeah, figured it was a feature request. :)

I've gone the first route, but in reality repositories dont quite have what I need yet because of more advanced querying needs. When I've got more time I'll take a pass at adding the support I need. :)

**Comment by elucash** [comment]
Yep, it should be possible and you could always resort to `find(String jsonCriteria)`  (not type safe, but workable). Feel free to ask questions about internals

**Comment by elucash** [comment]
`@Mongo.Repository` could now be used with `@Value.Include`, Released in 2.1.1. Thank you!
